src/libraries/Common/src/System/Security/Cryptography/X509Certificates/X509CertificateLoader.Pkcs12.cs

```diff
@@ -0,0 +1,909 @@
+                                // If this has a LocalKeyId and something references it, then it'll fail.
+
+                                continue;
+                            }
+                        }
+
+                        ConfirmPassword();
+
+                        Debug.Assert(decrypted.Array is not null);
+                        Debug.Assert(_keyDecryptBuffer.Length - _keyDecryptBufferOffset >= decrypted.Count);
+                        decrypted.AsSpan().CopyTo(_keyDecryptBuffer.AsSpan(_keyDecryptBufferOffset));
+
+                        ReadOnlyMemory<byte> newBagValue = new(
+                            _keyDecryptBuffer,
+                            _keyDecryptBufferOffset,
+                            decrypted.Count);
+
+                        CryptoPool.Return(decrypted);
+                        _keyDecryptBufferOffset += newBagValue.Length;
+
+                        if (contentRead != bag.BagValue.Length)
+                        {
+                            throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                        }
+
+                        bag.BagValue = newBagValue;
+                        bag.BagId = Oids.Pkcs12KeyBag;
+                    }
+                }
+            }
+
+            internal ArraySegment<byte> ToPfx(ReadOnlySpan<char> password)
+            {
+                Debug.Assert(_certBags is not null);
+                Debug.Assert(_keyBags is not null);
+
+                ContentInfoAsn safeContents = new ContentInfoAsn
+                {
+                    ContentType = Oids.Pkcs7Data,
+                };
+
+                AsnWriter writer = new AsnWriter(AsnEncodingRules.BER);
+                writer.PushOctetString();
+                writer.PushSequence();
+
+                for (int i = 0; i < _certCount; i++)
+                {
+                    SafeBagAsn bag = _certBags[i];
+                    bag.Encode(writer);
+                }
+
+                for (int i = 0; i < _keyCount; i++)
+                {
+                    SafeBagAsn bag = _keyBags[i];
+                    bag.Encode(writer);
+                }
+
+                writer.PopSequence();
+                writer.PopOctetString();
+                safeContents.Content = writer.Encode();
+                writer.Reset();
+
+                writer.PushSequence();
+                safeContents.Encode(writer);
+                writer.PopSequence();
+                byte[] authSafe = writer.Encode();
+                writer.Reset();
+
+                HashAlgorithmName hashAlgorithm = HashAlgorithmName.SHA1;
+                byte[] macKey = new byte[20];
+                Span<byte> salt = stackalloc byte[macKey.Length];
+
+#if NETCOREAPP3_1_OR_GREATER || NETSTANDARD2_1_OR_GREATER
+                RandomNumberGenerator.Fill(salt);
+#else
+                // "Randomly" produce the all zero salt.
+                salt.Clear();
```

Comment:
What’s the thinking here with an all zero salt instead of allocating a RNGCryptoServiceProvider and an array, and assigning that to salt?

Reply:
Now that I've actually hooked up a .NET Standard build this uses a static RNG instance instead of just Clear.